flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartPipelinedRegionStrategyTest.java

```diff
@@ -30,37 +38,245 @@
 public class RestartPipelinedRegionStrategyTest extends TestLogger {
```

Comment:
What I'm missing here are a few _minimal_ tests for the key backtracking behaviors, ideally with similar topologies to reduce cognitive load.

For example,

```
v1 |-\
      -> v3	  
v2 |-/

v1/v2  fails -> v1&v3 is restarted, v2 optionally depending on partition state
v3 fails, v3 is restarted, v1/v2 optionally dependending on partition state
```